beets/importer.py

```diff
@@ -520,6 +523,34 @@ def remove_duplicates(self, lib):
                 util.prune_dirs(os.path.dirname(item.path),
                                 lib.directory)
 
+    def merge_duplicates(self, lib):
+        """Combine an import with an existing album.
+
+        Used after resolve_duplicates determines that there are
+        duplicates and the user selects merge.  Both sets of items have
+        already been added to the db; so this function combines them
+        into one list, zeroes out their ids, removes them from the db,
+        and re-adds them as a single album.
+        """
+
+        # Tag all with the same album id
+        existing_album = self.find_duplicates(lib)[0]
+        for item in self.items:
+            item.album_id = existing_album.id
+
+        # Must set album_id before setting choice_flag
+        # choice_flag used for self.import_items() in other methods
+        self.choice_flag = action.ASIS
```

Comment:
To check my understanding: It looks like this version of the feature doesn't *re-tag* the new, combined album. Instead, the match for the old album is ignored, and the files keep their current metadata. Is that right?

We may want to explore the other thing, which sounds like what people were originally interested in: the auto-tagger gets to "try again" with the new, larger album.